components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -3777,4 +3777,65 @@ public static String getMountedPath(RegistryContext registryContext, String path
         }
         return domains;
     }
+
+    /**
+     * This method used to Downloaded Uploaded Documents from publisher
+     *
+     * @param userName     logged in username
+     * @param resourceUrl  resource want to download
+     * @param tenantDomain loggedUserTenantDomain
+     * @return map that contains Data of the resource
+     * @throws APIManagementException
+     */
+
+    public static Map<String, Object> getDocument(String userName, String resourceUrl,
+                                                  String tenantDomain)
+            throws APIManagementException {
+        Map<String, Object> documentMap = new HashMap<String, Object>();
+
+        InputStream inStream = null;
+        String[] resourceSplitPath =
+                resourceUrl.split(RegistryConstants.GOVERNANCE_REGISTRY_BASE_PATH);
+        if (resourceSplitPath.length == 2) {
+            resourceUrl = resourceSplitPath[1];
+        } else {
+            handleException("Invalid resource Path " + resourceUrl);
+        }
+        Resource apiDocResource;
+        Registry registryType = null;
+        int tenantId = MultitenantConstants.SUPER_TENANT_ID;
+        try {
+            if (tenantDomain != null && !"null".equals(tenantDomain)) {
+                tenantId = ServiceReferenceHolder
+                        .getInstance().getRealmService().getTenantManager()
+                        .getTenantId(tenantDomain);
+            }
+            if (tenantId != MultitenantConstants.SUPER_TENANT_ID) {
+                userName = userName.split("@" + tenantDomain)[0];
```

Comment:
Using MultitenantUtils.getTenantAwareUsername rather than splitting username with '@'. In case email username is enabled, there are several combinations that needs to be handled and when username is just split using '@', we might fail to consider all those different scenarios.